x/poe/module.go

```diff
@@ -164,12 +165,13 @@ func (am AppModule) EndBlock(ctx sdk.Context, block abci.RequestEndBlock) []abci
 func (am AppModule) InitGenesis(ctx sdk.Context, cdc codec.JSONCodec, data json.RawMessage) []abci.ValidatorUpdate {
 	var genesisState types.GenesisState
 	cdc.MustUnmarshalJSON(data, &genesisState)
-	if len(genesisState.GenTxs) == 0 {
-		panic(sdkerrors.Wrap(wasmtypes.ErrInvalidGenesis, "empty gentx"))
-	}
 
-	if genesisState.SeedContracts {
-		if err := BootstrapPoEContracts(ctx, am.contractKeeper, am.twasmKeeper, am.poeKeeper, genesisState); err != nil {
+	seedMode := genesisState.SeedContracts != nil
+	if seedMode {
+		if len(genesisState.SeedContracts.GenTxs) == 0 {
+			panic(sdkerrors.Wrap(wasmtypes.ErrInvalidGenesis, "empty gentx"))
+		}
+		if err := BootstrapPoEContracts(ctx, am.contractKeeper, am.twasmKeeper, am.poeKeeper, *genesisState.SeedContracts); err != nil {
 			panic(fmt.Sprintf("bootstrap PoE contracts: %+v", err))
 		}
 	}
```

Comment:
We could then avoid passing the entire genesisState to InitGenesis, and just the Params that is used in both cases